src/Datasource/FactoryLocator.php

```diff
@@ -27,18 +27,18 @@ class FactoryLocator
     /**
      * A list of model factory functions.
      *
-     * @var callable[]
+     * @var (callable|\Cake\Datasource\LocatorInterface)[]
      */
     protected static $_modelFactories = [];
 
     /**
      * Register a callable to generate repositories of a given type.
      *
      * @param string $type The name of the repository type the factory function is for.
-     * @param callable $factory The factory function used to create instances.
+     * @param callable|\Cake\Datasource\LocatorInterface $factory The factory function used to create instances.
      * @return void
      */
-    public static function add(string $type, callable $factory): void
+    public static function add(string $type, $factory): void
```

Comment:
Should we do checks here to ensure that $factory is of an acceptable type?

Reply:
Done